beanie/odm/utils/encoder.py

```diff
@@ -26,6 +26,58 @@
 from beanie.odm.fields import Link, LinkTypes
 from beanie.odm.utils.pydantic import IS_PYDANTIC_V2, get_model_fields
 
+DT_utc_object = datetime.datetime.now(datetime.timezone.utc)
+
+
+def _compatct_time_encoder(time: datetime.time) -> str:
+    # reimpementation of speeddate in python  https://github.com/pydantic/speedate/blob/aad1d9117a05618ae883e20ae179c582eb998eeb
+    # /src/time.rs#L41 and pydantic https://github.com/pydantic/pydantic-core/blob/f636403c2e8169bcbb94aa71a0727d76076f7e6e/src/input/datetime.rs#L177
+    if time.microsecond != 0:
+        time_string = "{:02d}:{:02d}:{:02d}.{:06d}".format(
+            time.hour, time.minute, time.second, time.microsecond
+        )
+    elif time.second != 0:
+        time_string = "{:02d}:{:02d}:{:02d}".format(
+            time.hour, time.minute, time.second
+        )
+    else:
+        time_string = "{:02d}:{:02d}".format(time.hour, time.minute)
+
+    tz_offset = None
+    if time.tzinfo is not None:  # has timezone
+        # check if the offset is fixed no matter the datetime
+        offset_delta = time.tzinfo.utcoffset(None)
+        if offset_delta is None:  # is ambiguous resolve base on current time
```

Comment:
i am not shure if this is right... 
this may l change serialization based on date... 

This is caused when there is two variance of timezone  (two timedeltas) like Europe/Prague having summer and winter variant(DST) and its not conclusive what is right

alternatives:

1. throw error that timezone is inconclusive 
2. serialize to particular date (like 1.1.1970) but this ventures to defining a standard